### PR TITLE
Feature/cook recipe post

### PR DESCRIPTION
**Pull request by aliroemhildt**
#### What's this PR do?
- removes ingredient amounts from a user's pantry when they cook a recipe
- refactor api fetch calls: moved post request into
- refactor populate table function 
- add modifyIngredient method to user: the returns the object that is passed into the post fetch request
- add test for modifyIngredients user method

#### Where should the reviewer start?
- apiCalls file
- jumped all around in scripts

#### How should this be manually tested?
- the pantry page on the site,
- add and remove ingredients

### CODE REVIEW

File: src/classes/User.js [hunk 1]
```diff
@@ -63,6 +63,14 @@ class User {
       return acc;
     }, [])
   }
+
+  modifyIngredient(id, amount) {
```
Comment (anewb87) [inline src/classes/User.js:67]: We felt like putting this in the class would streamline/clean up our code.

File: src/apiCalls.js [hunk 2]
```diff
@@ -7,4 +7,14 @@ let ingredientAPI = fetch("http://localhost:3001/api/v1/ingredients")
 let recipeAPI = fetch("http://localhost:3001/api/v1/recipes")
   .then(response => response.json())
 
-export {userAPI, ingredientAPI, recipeAPI}
+let post = async (info) => {
```
Comment (anewb87) [inline src/apiCalls.js:10]: Moved this to the apiCalls file.

**Review by kammon10 — APPROVED** [review]

File: src/scripts.js [hunk 3]
```diff
@@ -362,9 +375,12 @@ function displayIngredientsNeeded(recipe) {
...
 
   if (neededIngredients.length === 0) {
-    neededIngredientsSection.innerHTML = `
-      <p>you have all of the ingredients needed to cook this recipe!</p>
-      <button>cook recipe</button>`;
+    neededIngredientsSection.innerHTML =
+      `<p>you have all of the ingredients needed to cook this recipe!</p>
+      <button class="cook-recipe-js">cook recipe</button>`;
+
+      const cookRecipeButton = document.querySelector('.cook-recipe-js');
```
Comment (anewb87) [inline src/scripts.js:382]: Our button officially posts (deletes) the recipe ingredients from the pantry.

File: src/scripts.js [hunk 4]
```diff
@@ -383,6 +399,20 @@ function displayIngredientsNeeded(recipe) {
   }
 }
 
+async function removeRecipeIngredients() {
+  for (const ingredient of selectedRecipe.ingredients) {
```
Comment (anewb87) [inline src/scripts.js:403]: We were struggling with trying to put await in a forEach. We aren't quite understanding exactly why (and we're thinking we might try to still re-work this await), but using this syntax works.

File: src/scripts.js [hunk 5]
```diff
@@ -383,6 +399,20 @@ function displayIngredientsNeeded(recipe) {
...
+    const amount = -(parseFloat(ingredient.quantity.amount));
+    // await postToPantry(ingredient.id, amount);
+    const info = currentUser.modifyIngredient(ingredient.id, amount);
+    await postToPantry(info);
+  }
+}
+
+async function removeIngredients() {
```
Comment (anewb87) [inline src/scripts.js:411]: A function of helper functions!

File: src/scripts.js [hunk 6]
```diff
@@ -496,7 +527,11 @@ function getIngredientListElement(e, selectedRecipe) {
   ingredientListSection.innerHTML = '<h3>ingredients</h3>';
   const ingredientNames = selectedRecipe.determineRecipeIngredients(ingredientsData);
   const ingredientListText = selectedRecipe.ingredients.reduce((acc, ingredient, index) => {
-    acc += `${ingredientNames[index]}: ${ingredient.quantity.amount.toString()} ${ingredient.quantity.unit}<br><br>`;
+    let amount = ingredient.quantity.amount
+    if (!(ingredient.quantity.amount % 1) === 0) {
```
Comment (anewb87) [inline src/scripts.js:531]: We didn't like having infinitely long decimals, so added this in to clean it up.

File: test/User-test.js [hunk 7]
```diff
@@ -72,4 +72,16 @@ describe('User', () => {
 
     expect(firstUser.filterFavoritesByName('vanilla')).to.deep.equal([sampleData[0], sampleData[5]]);
   });
+
+  it('should return ingredient information as an object, given an ingredient id and amount', () => {
```
Comment (anewb87) [inline test/User-test.js:76]: The test we needed to implement since we added a method to our User class.

**Review by anewb87 — APPROVED** [review]
Checking stuff off the list!

**Review by cagallo — COMMENTED** [review]
looks good!

**Pull request merged**